coordinator/core/clientapi.go

```diff
@@ -184,7 +221,17 @@ func (c *Core) UpdateManifest(ctx context.Context, rawUpdateManifest []byte) err
 
 	c.updateManifest = updateManifest
 	c.rawUpdateManifest = rawUpdateManifest
+	c.intermediateCert = intermediateCert
+	c.intermediatePrivK = intermediatePrivK
+
+	// Overwrite regenerated secrets in core
+	for name, secret := range regeneratedSecrets {
+		delete(c.secrets, name)
```

Comment:
seems superfluous to me

Reply:
Actually I had to introduce it as a bug fix. Directly assigning the secret to the map in the core and thus, wanting to simply overwrite it, did not work on its own. 

Don't know why, but would be interesting to find out why it does not work. The most common solution you find online is using pointers in the map as value rather than a copy to the struct directly, but personally I would try to avoid using this solution as we do not really want to modify the secrets in the core often, just out of necessity on a manifest update.

Reply:
Works fine on my machine if I remove this line. Can you double-check and tell which test fails?

Reply:
Double-checked, it actually does work. 

Guess this was an issue in the test then, as I did not deep-copy the map before comparing them initially, so I guess I was comparing pointers instead of the values in the structs, thus having this weird issue where delete fixed it. Later on I noticed this caused some other issues, so I deep-copied the map in the test which seems to fix all the issues I had.

So yeah, this can be removed.